blockdag/validate.go

```diff
@@ -717,11 +706,20 @@ func (dag *BlockDAG) validateParents(blockHeader *domainmessage.BlockHeader, par
 // for how the flags modify its behavior.
 //
 // This function MUST be called with the dag state lock held (for writes).
-func (dag *BlockDAG) checkBlockContext(block *util.Block, parents blockSet, flags BehaviorFlags) error {
+func (dag *BlockDAG) checkBlockContext(block *util.Block, flags BehaviorFlags) error {
+	err := dag.checkBlockParents(block)
+	if err != nil {
+		return err
+	}
+	parents, err := lookupParentNodes(block, dag)
```

Comment:
You just did lookupParentNodes in checkBlockParents. 
Why not return it from there and use here?